src/components/Common/AddressConn.js

```diff
@@ -16,6 +16,28 @@ const AddressConn = () => {
 
   return (
     <>
+      {!wallet && (
+        <Button
+          type="button"
+          className={btnClass}
+          onClick={() => setWalletModalShow(true)}
+          onKeyPress={() => setWalletModalShow(true)}
+        >
+          <i className={`icon-wallet-red${iconClass}`} />
+        </Button>
+      )}
+
+      {wallet?.status === 'connecting' && (
+        <Button
+          type="button"
+          className={btnClass}
+          onClick={() => setWalletModalShow(true)}
+          onKeyPress={() => setWalletModalShow(true)}
+        >
+          <i className={`icon-wallet-red${iconClass}`} />
+        </Button>
+      )}
+
```

Comment:
Added more state checks for wallet to ensure wallet icon is always showing (and only once; no duplicates)